### PR TITLE
change addr prefix to "atest1"

### LINKED ISSUE #478: switch to "atest" prefix for addresses
in `shared/src/types/address.rs`: the bech32m address encoding prefix `ADDRESS_HRP` should be "atest" for "dev" and testnets. We should only use "a" for mainnet, once we're there. We'll need to update all the default addresses for this (replace `s/a1/atest1`)

### PR DESCRIPTION
**Pull request by tzemanovic**
closes #478 and #491

I removed borsh encoding usage from address bech32m encoding and replaced with a fixed length utf-8 strings as bytes, so that all the address types are equal length (84 with prefix "atest1", 80 with "a1").

I also switched `make install` to release build without "dev" feature.

### CODE REVIEW

File: shared/src/types/address.rs [hunk 1]
```diff
@@ -14,17 +14,41 @@ use sha2::{Digest, Sha256};
...
+// TODO use "a" for live network
+const ADDRESS_HRP: &str = "atest";
 const ADDRESS_BECH32_VARIANT: bech32::Variant = Variant::Bech32m;
 pub(crate) const HASH_LEN: usize = 40;
 
+/// An address string before bech32m encoding must be this size.
+pub const FIXED_LEN_STRING: usize = 45;
+
+/// Raw strings used to produce internal addresses. All the strings must begin
+/// with `PREFIX_INTERNAL` and be `FIXED_LEN_STRING` characters long.
```
Comment (juped) [inline shared/src/types/address.rs:35]: should align these in the source file so it's clear they're the same length

File: shared/src/types/address.rs [hunk 2]
```diff
@@ -14,17 +14,41 @@ use sha2::{Digest, Sha256};
...
+mod internal {
+    pub const POS: &str = "ano::Proof of Stake                          ";
+    pub const POS_SLASH_POOL: &str =
+        "ano::Proof of Stake Slash Pool               ";
```
Comment (juped) [inline shared/src/types/address.rs:39]: (do them all like this one)

File: shared/src/types/address.rs [hunk 3]
```diff
@@ -131,6 +137,73 @@ impl Address {
         }
     }
 
+    /// Convert an address to a fixed length utf-8 string bytes
```
Comment (juped) [inline shared/src/types/address.rs:140]: 7-bit ascii?

File: shared/src/types/address.rs [hunk 4]
```diff
@@ -131,6 +137,73 @@ impl Address {
...
+            }
+            Address::Internal(internal) => match internal {
+                InternalAddress::PoS => internal::POS,
+                InternalAddress::PosSlashPool => internal::POS_SLASH_POOL,
+                InternalAddress::Ibc => internal::IBC,
+                InternalAddress::Parameters => internal::PARAMETERS,
+            }
+            .to_string(),
+        }
+        .into_bytes()
```
Comment (juped) [inline shared/src/types/address.rs:157]: an assertion that it's FIXED_LEN_STRING bytes might be nice

File: shared/src/types/address.rs [hunk 5]
```diff
@@ -14,17 +14,41 @@ use sha2::{Digest, Sha256};
...
 
 /// human-readable part of Bech32m encoded address
-const ADDRESS_HRP: &str = "a";
+// TODO use "a" for live network
+const ADDRESS_HRP: &str = "atest";
 const ADDRESS_BECH32_VARIANT: bech32::Variant = Variant::Bech32m;
 pub(crate) const HASH_LEN: usize = 40;
 
+/// An address string before bech32m encoding must be this size.
+pub const FIXED_LEN_STRING: usize = 45;
```
Comment (juped) [inline shared/src/types/address.rs:32]: FIXED_LEN_STRING_BYTES?

File: shared/src/types/address.rs (hunk 4 above)
Comment (tzemanovic) [reply shared/src/types/address.rs]: Good idea, I've added debug assert here. There' a test for this `test_encoded_address_length` that checks all the addresses.

File: shared/src/types/address.rs (hunk 2 above)
Comment (tzemanovic) [reply shared/src/types/address.rs:41]: done

File: shared/src/types/address.rs (hunk 3 above)
Comment (tzemanovic) [reply shared/src/types/address.rs]: changed, thanks!

**Pull request merged**